pkg/advancedapi/advancedApi.go

```diff
@@ -25,65 +25,41 @@ func GetRegisterDocument(resolverClient register.ResolverClient, did string) (*r
 }
 
 // RegisterNewDocument Create and register a new document against the resolver.
-func RegisterNewDocument(resolverClient register.ResolverClient, keyPair *crypto.KeyPair, purpose identity.DidType, name string, override bool) error {
-	did, err := identity.MakeIdentifier(keyPair.PublicKeyBytes)
+func RegisterNewDocument(resolverClient register.ResolverClient, keyPair *crypto.KeyPair, purpose identity.DidType, name string, override bool) (*register.RegisterDocument, error) {
+	registerDocument, issuer, err := NewDocument(purpose, keyPair, name)
 	if err != nil {
-		return err
+		return nil, err
 	}
+
 	if !override {
-		getDoc, _ := GetRegisterDocument(resolverClient, did)
+		getDoc, _ := GetRegisterDocument(resolverClient, issuer.Did)
```

Comment:
Shouldn't this return the error, if there is one? (Otherwise in the case `/discover` failed - it's as if `override` was set - not sure that is intended.)